### PR TITLE
turn-off checking for updates completely if MINIO_UPDATE=off

### LINKED ISSUE #10723: Long timeout connecting to https://dl.min.io before run process
Thank you for closed my issue #10674  without debug and research. Way to go guys!

When I trying run simple server like "./minio server /data", the process trying resolve name dl.min.io before start and trying connect to it(WHY???). My server can resolve all DNS records, but doesn't have access to Internet. As a result, I have to wait about 20 min until happens a timeout. 

tcpdump output:
    10.70.31.15.53 > 10.70.31.196.55174: [udp sum ok] 40353 q: A? dl.min.io. 1/4/8 dl.min.io. [3m43s] A 178.128.69.202 ns: min.io. [19h16m54s] NS ns-cloud-e3.googledomains.com., min.io. [19h16m54s] NS ns-cloud-e1.googledomains.com., min.io. [19h16m54s] NS ns-cloud-e4.googledomains.com., min.io. [19h16m54s] NS ns-cloud-e2.googledomains.com. ar: ns-cloud-e1.googledomains.com. [7h12m42s] A 216.239.32.110, ns-cloud-e1.googledomains.com. [19h41m27s] AAAA 2001:4860:4802:32::6e, ns-cloud-e4.googledomains.com. [1d9h30m46s] A 216.239.38.110, ns-cloud-e4.googledomains.com. [19h41m27s] AAAA 2001:4860:4802:38::6e, ns-cloud-e3.googledomains.com. [22h9m35s] A 216.239.36.110, ns-cloud-e3.googledomains.com. [19h41m27s] AAAA 2001:4860:4802:36::6e, ns-cloud-e2.googledomains.com. [22h39m34s] A 216.239.34.110, ns-cloud-e2.googledomains.com. [19h41m27s] AAAA 2001:4860:4802:34::6e (340)
    10.70.31.196.58936 > 178.128.69.202.443: Flags [P.], cksum 0x2338 (incorrect -> 0x59ee), seq 0:189, ack 1, win 229, options [nop,nop,TS val 2590971309 ecr 3918460386], length 189
 00:00:00.188412 IP (tos 0x0, ttl 64, id 44881, offset 0, flags [DF], proto TCP (6), length 52)
    10.70.31.196.58936 > 178.128.69.202.443: Flags [.], cksum 0x227b (incorrect -> 0x0dfa), seq 189, ack 4204, win 294, options [nop,nop,TS val 2590971497 ecr 3918460673], length 0
 00:00:00.005087 IP (tos 0x0, ttl 64, id 44882, offset 0, flags [DF], proto TCP (6), length 145)
----- many are the same strings ------

If DNS has disabled or has set fake env "http_proxy", like "http_proxy="fake" ./minio server /data", this problem is fixed.

### PR DESCRIPTION
**Pull request by harshavardhana**
## Description
turn-off checking for updates completely if MINIO_UPDATE=off

## Motivation and Context
fixes #10723 

## How to test this PR?
Nothing much just check that we don't see requests to dl.min.io when MINIO_UPDATE=off

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Documentation needed
- [ ] Unit tests needed

### CODE REVIEW

**Review by kannappanr — APPROVED** [review]
LGTM

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-zoned.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :x: [more...](#fbfa5d14-d432-4761-96f3-a460a6eb1b88) |
<a name="fbfa5d14-d432-4761-96f3-a460a6eb1b88"></a>
### 10752-e94cc0d/mint-gateway-azure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-c3.minio.io:30814
ACCESS_KEY:           minioazure
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 706d9d75acea:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 8 seconds
(2/15) Running aws-sdk-java tests ... done in 2 seconds
(3/15) Running aws-sdk-php tests ... done in 2 minutes and 25 seconds
(4/15) Running aws-sdk-ruby tests ... done in 20 seconds
(5/15) Running awscli tests ... done in 2 minutes and 50 seconds
(6/15) Running healthcheck tests ... done in 1 seconds
(7/15) Running mc tests ... done in 3 minutes and 22 seconds
(8/15) Running minio-dotnet tests ... done in 1 minutes and 39 seconds
(9/15) Running minio-go tests ... FAILED in 1 minutes and 53 seconds
{
  "args": {
    "bucketName": "minio-go-test-xy2oavm9q6x0eido",
    "objectName": "test-object",
    "opts": "",
    "size": -1
  },
  "duration": 1274,
  "function": "PutObject(bucketName, objectName, reader,size,opts)",
  "message": "Unexpected size",
  "name": "minio-go: testPutObjectStreaming",
  "status": "FAIL"
}
(9/15) Running minio-java tests ... done in 10 minutes and 42 seconds
(10/15) Running minio-js tests ... FAILED in 47 seconds
{
  "name": "minio-js",
  "function": "\"after all\" hook in \"functional tests\"",
  "duration": 81,
  "status": "FAIL",
  "error": "S3Error: The bucket you tried to delete is not empty at Object.parseError (node_modules/minio/dist/main/xml-parsers.js:86:11) at /mint/run/core/minio-js/node_modules/minio/dist/main/transformers.js:156:22 at DestroyableTransform._flush (node_modules/minio/dist/main/transformers.js:80:10) at DestroyableTransform.prefinish (node_modules/readable-stream/lib/_stream_transform.js:129:10) at prefinish (node_modules/readable-stream/lib/_stream_writable.js:611:14) at finishMaybe (node_modules/readable-stream/lib/_stream_writable.js:620:5) at endWritable (node_modules/readable-stream/lib/_stream_writable.js:643:3) at DestroyableTransform.Writable.end (node_modules/readable-stream/lib/_stream_writable.js:571:22) at IncomingMessage.onend (_stream_readable.js:682:10) at endReadableNT (_stream_readable.js:1252:12) at processTicksAndRejections (internal/process/task_queues.js:80:21)"
}
(10/15) Running minio-py tests ... done in 18 minutes and 32 seconds
(11/15) Running s3cmd tests ... done in 2 minutes and 17 seconds
(12/15) Running s3select tests ... done in 38 seconds
(13/15) Running security tests ... done in 0 seconds

Executed 13 out of 15 tests successfully.
```
Deleting image on docker hub
Deleting image locally

File: cmd/server-main.go [hunk 1]
```diff
@@ -418,7 +418,7 @@ func serverMain(ctx *cli.Context) {
 		}
 	}
 
-	if !globalCLIContext.Quiet {
+	if !globalCLIContext.Quiet && globalInplaceUpdateDisabled {
```
Comment (rafi) [inline cmd/server-main.go:421]: @harshavardhana shouldn't this be a negation?  Looks like `MINIO_UPDATE=on` disables the update now, and not `off`.

Comment (rafi) [reply cmd/server-main.go:421]: Created PR #10900

Comment (harshavardhana) [reply cmd/server-main.go:421]: Correct @rafi

**Pull request merged**